From 62f510702ef9c34ac50f8d8601a4290ab558464c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?=C3=96mer=20Fad=C4=B1l=20Usta?= <omerusta@gmail.com>
Date: Sun, 6 Jun 2021 08:41:05 +0000
Subject: [PATCH] Update Cmake and deps, Fix Freetype and FontConfig Linkage

Cmake to 3.16
KF to 5.76
QT to 5.12
ECM follow KF's version
Update CALLIGRA_YEAR to 2021
Fix Fontconfig's FOUND variable
Fix Fontconfig's include_dirs variable
Use Correct Target to link FontConfig and Freetype
---
 CMakeLists.txt           | 20 +++++++-------------
 libs/text/CMakeLists.txt |  4 ++--
 2 files changed, 9 insertions(+), 15 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b237f68fcb6..cdcc02f0236 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.2)
+cmake_minimum_required(VERSION 3.16)
 
 project(calligra)
 
@@ -35,7 +35,7 @@ set(CALLIGRA_VERSION_RELEASE 89)     # 89 for Alpha, increase for next test rele
 set(CALLIGRA_ALPHA 1) # uncomment only for Alpha
 # set(CALLIGRA_BETA 1) # uncomment only for Beta
 #set(CALLIGRA_RC 1) # uncomment only for RC
-set(CALLIGRA_YEAR 2020) # update every year
+set(CALLIGRA_YEAR 2021) # update every year
 
 if(NOT DEFINED CALLIGRA_ALPHA AND NOT DEFINED CALLIGRA_BETA AND NOT DEFINED CALLIGRA_RC)
     set(CALLIGRA_STABLE 1) # do not edit
@@ -89,9 +89,6 @@ if(NOT DEFINED RELEASE_BUILD)
 endif()
 message(STATUS "Release build: ${RELEASE_BUILD}")
 
-# use CPP-11
-set (CMAKE_CXX_STANDARD 11)
-
 ############
 #############
 ## Options ##
@@ -148,8 +145,10 @@ calligra_set_productset(${PRODUCTSET})
 ## Look for ECM, Qt, KF5 ##
 ###########################
 ##########################
+set(REQUIRED_KF5_VERSION "5.76.0")
+set(REQUIRED_QT_VERSION "5.12.0")
 
-find_package(ECM 5.19 REQUIRED NO_MODULE)
+find_package(ECM ${REQUIRED_KF5_VERSION} REQUIRED NO_MODULE)
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${ECM_MODULE_PATH} ${ECM_KDE_MODULE_DIR})
 
 # ECM KDE macros (include first, to have their policies and settings effect all other macros)
@@ -179,9 +178,6 @@ include(MacroOptionalFindPackage)
 include(MacroEnsureVersion)
 include(MacroDesktopToJson)
 
-
-set(REQUIRED_KF5_VERSION "5.30.0")
-
 find_package(KF5 ${REQUIRED_KF5_VERSION} REQUIRED
     COMPONENTS
     Archive
@@ -224,8 +220,6 @@ if(KF5Activities_FOUND)
     set(HAVE_KACTIVITIES TRUE)
 endif()
 
-set(REQUIRED_QT_VERSION "5.6.0")
-
 find_package(Qt5 ${REQUIRED_QT_VERSION} REQUIRED
     COMPONENTS
     Core
@@ -599,8 +593,8 @@ if(NOT WIN32 AND NOT APPLE)
 	)
 endif()
 
-if(NOT FONTCONFIG_FOUND OR NOT FREETYPE_FOUND)
-    set(FONTCONFIG_INCLUDE_DIR "")
+if(NOT Fontconfig_FOUND OR NOT FREETYPE_FOUND)
+    set(Fontconfig_INCLUDE_DIRS "")
     set(FREETYPE_INCLUDE_DIRS "")
 else()
     add_definitions( -DSHOULD_BUILD_FONT_CONVERSION )
diff --git a/libs/text/CMakeLists.txt b/libs/text/CMakeLists.txt
index 16d7e066fa2..429afe79704 100644
--- a/libs/text/CMakeLists.txt
+++ b/libs/text/CMakeLists.txt
@@ -152,11 +152,11 @@ if( SHOULD_BUILD_FEATURE_RDF )
 endif()
 
 if( FONTCONFIG_FOUND )
-    target_link_libraries(kotext PRIVATE ${FONTCONFIG_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Fontconfig::Fontconfig)
 endif()
 
 if( FREETYPE_FOUND )
-    target_link_libraries(kotext PRIVATE ${FREETYPE_LIBRARIES})
+    target_link_libraries(kotext PRIVATE Freetype::Freetype)
 endif()
 
 
-- 
GitLab

